### PR TITLE
Feat/solar2colmap

### PR DESCRIPTION
**Pull request by akacete**

### CODE REVIEW

**Comment by granger35** [comment]
LGTM

**Review by granger35 — APPROVED** [review]
Ok for merging in master branch.

File: SolARFramework.pri [hunk 1]
```diff
@@ -1,4 +1,5 @@
 HEADERS += interfaces/api/display/I2DOverlay.h \
+    $$PWD/interfaces/api/output/files/IMapExporter.h \
```
Comment (jim-bcom) [inline SolARFramework.pri:2]: Inconsistent path style

File: interfaces/api/output/files/IMapExporter.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#ifndef IMAPEXPORTER_H
```
Comment (jim-bcom) [inline interfaces/api/output/files/IMapExporter.h:1]: Add listing header

File: interfaces/api/output/files/IMapExporter.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ */
+class XPCF_IGNORE IMapExporter :
+                                  virtual public org::bcom::xpcf::IComponentIntrospect {
+public:
+    IMapExporter() = default;
+    virtual ~IMapExporter() = default;
+
+    /// @brief Export a Map to a file
+    /// @param[in] map: the map to export
```
Comment (jim-bcom) [inline interfaces/api/output/files/IMapExporter.h:26]: Remove ":"

File: interfaces/api/output/files/IMapExporter.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    /// @return FrameworkReturnCode::_SUCCESS if export succeed, else FrameworkReturnCode::_ERROR_
+    virtual FrameworkReturnCode exportMap(const SRef<SolAR::datastructure::Map> & map) = 0;
```
Comment (jim-bcom) [inline interfaces/api/output/files/IMapExporter.h:28]: Could it be `const SRef<const SolAR::datastructure::Map> & map` ?

**Review by jim-bcom — COMMENTED** [review]
Sorry for the unsollicited comments.

**Pull request merged**